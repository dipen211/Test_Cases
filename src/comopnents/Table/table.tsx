import React, { Component } from 'react';
class Table extends Component {
    state = {
        items: [] as any,
        item: '',
    };
    onChange = (e: any) => {
        this.setState({
            item: e.target.value,
        });
    };
    addItem = (e: any) => {
        e.preventDefault();
        this.setState({
            items: this.state.items.concat(
                this.state.item
            ),
            item: '',
        });
    };
    render() {
        const submitDisabled = !this.state.item;
        return (
            <>
                <form onSubmit={this.addItem}>
                    <div>
                        <input
                            className='prompt'
                            type='text'
                            placeholder='Add item...'
                            value={this.state.item}
                            onChange={this.onChange}
                        />
                    </div>
                    <button
                        type='submit'
                        disabled={submitDisabled}
                    >
                        Add item
                    </button>
                </form>
                <table style={{ margin: "auto" }}>
                    <thead>
                        <tr>
                            <th>Items</th>
                        </tr>
                    </thead>
                    <tbody>
                        {
                            this.state.items.map((item: any, id: any) => (
                                <tr
                                    key={id}
                                >
                                    <td>{item}</td>
                                </tr>
                            ))
                        }
                    </tbody>
                </table>
            </>
        )
    }
}
export default Table;