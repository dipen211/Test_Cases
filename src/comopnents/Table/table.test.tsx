import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Table from '../../comopnents/Table/table';

Enzyme.configure({ adapter: new Adapter() });
describe('Table component tests', () => {
    let wrapper: Enzyme.ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
    beforeEach(() => {
        wrapper = shallow(<Table />);
    });

    it('should have the <th> Items', () => {
        expect(wrapper.contains(<th>Items</th>)).toBe(true);
    });
    it('should have a <button> element', () => {
        expect(wrapper.containsMatchingElement(<button>Add item</button>)).toBe(true);
    });
    it('<button> should be disabled', () => {
        const button = wrapper.find('button').first(); expect(button.props().disabled).toBe(true);
    });
    describe('input component tests', () => {
        let item = 'Laundry';
        beforeEach(() => {
            const input = wrapper.find('input').first();
            input.simulate('change', {
                target: { value: item }
            })
        });
        it('should update the state property <item>', () => {
            expect(wrapper.state().item).toEqual(item);
        });
        it('should enable <button>', () => {
            const button = wrapper.find('button').first();
            expect(
                button.props().disabled
            ).toBe(false);
        });
        describe('and then clears the input', () => {
            beforeEach(() => {
                const input = wrapper.find('input').first();
                input.simulate('change', {
                    target: { value: '' }
                })
            });
            it('should disable <button>', () => {
                const button = wrapper.find('button').first();
                expect(
                    button.props().disabled
                ).toBe(true);
            });
        });
        describe('submit the form component test', () => {
            beforeEach(() => {
                const form = wrapper.find('form').first();
                form.simulate('submit', {
                    preventDefault: () => { },
                });
            });
            it('should add the item to state', () => {
                expect(
                    wrapper.state().items
                ).toContain(item);
            });
            it('should render the item in the table', () => {
                expect(
                    wrapper.containsMatchingElement(
                        <td>{item}</td>
                    )
                ).toBe(true);
            });
            it('should clear the input field', () => {
                const input = wrapper.find('input').first();
                expect(
                    input.props().value
                ).toEqual('');
            });
            it('should disable `button`', () => {
                const button = wrapper.find('button').first();
                expect(
                    button.props().disabled
                ).toBe(true);
            });
        });
    });
});