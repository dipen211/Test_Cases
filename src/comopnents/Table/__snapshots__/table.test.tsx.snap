// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component tests should match the snapshot 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <div>
      <input
        className="prompt"
        onChange={[Function]}
        placeholder="Add item..."
        type="text"
        value=""
      />
    </div>
    <button
      disabled={true}
      type="submit"
    >
      Add item
    </button>
  </form>
  <table
    style={
      Object {
        "margin": "auto",
      }
    }
  >
    <thead>
      <tr>
        <th>
          Items
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</Fragment>
`;
